content/post/2019-01-20-what-is-production-grade.markdown

```diff
@@ -0,0 +1,270 @@
+infrastructure software is written in Golang, as it has minimal
+assumptions about how much of the rest of the world around it is
+functional.
+
+### Subminiature
+
+Choosing software that runs on my home network is perhaps the most
+picky I can be.
+
+In many cases, this is software running without redundancy on
+non-redundant hardware.  Add to this that its running in what most
+people would consider to be a "privileged" environment and its easy to
+understand why I would be so picky.
+
+For example, my home network is powered by OpenBSD.  I use OpenBSD
+here because it has a history of being incredibly robust, very well
+built and documented, and it has a thriving community around it for if
+I ever do manage to find a real bug in it.  The only bits of my home
+router that are even remotely specialized are a pair of well supported
+network cards, and an IDE disk on module with industrial flash to keep
+the system quieter and lower power draw than a proper spinning disk
+would be.
+
+Another thing I'm looking for at this scale is that the software works
+without too much input.  Running OpenLDAP at a medium scale and above
+justifies its upkeep requirement by the features it provies, but
+running it in smaller environments doesn't make sense because of the
+mental complexity that it takes to operate.  At the subminiature scale
+of my home network, I'm truly looking for the system to be as low
+maintenence as possible.
+
+
+## To the Other Side
+
+"But wait!" I hear you say to your monitor, "I write software, not run
+other people's software like you!  How do I know that what I've
+written is ready to hand off to an operations team to deploy it and
+run it!?"
+
+This is a question I don't have an answer to.
+
+For over a year now I've been working on a project called NetAuth.
+Right now it satisfies all the requirements I have for running in the
+small to medium production system, but I'm still uncomfortable
+deploying it in prod.  Partially because it hasn't satisfied the
+many-eyes component that makes open source security softare good, but
+partially because I'm just not confident enough in my own software to
+say "yes, this is good, you should run it".
+
+Personally if you ever find an engineer that has that kind of attitude
+towards softare you should think very carefully about the claims they
+make, and find out who's head is on a platter if they're wrong.
+
+Interestingly, I would have no problems running Void Linux in prod.
+I've actually done this in the past, and my reasoning from then stands
+now.  I believe firmly that it is better to run a less battle hardened
+system that you can fit in entirely in your head rather than an
+"industry standard" one that you can't.  Anyone who's had to
+troubleshoot a large complex system during an outage will know exactly
+why this is the case.
+
+
+## TL;DR
+
+I think posts like this always need a takeaway that you can think on
+later.  To distill down all the ideas that are in this post into
```

Comment:
"To distill down all the ideas that are in this post into something you can chew on later" doesn't add anything.